### PR TITLE
Added Txnsync Service Unit Tests

### PR DESCRIPTION
**Pull request by AlgoStephenAkiki**
Added unit tests and fixed some spelling errors

### CODE REVIEW

File: txnsync/service_test.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	hashDigest := crypto.Hash([]byte{0x41, 0x6b, 0x69, 0x6b, 0x69})
+
+	service := MakeTransactionSyncService(mLogger, &mNodeConnector, true, "GENID", hashDigest, cfg, mThreadPool)
+
+	a.NotNil(service)
+
+	service.Start()
+	service.Stop()
+
+	a.True(calledEventsInNodeConnector)
```
Comment (tsachiherman) [inline txnsync/service_test.go:91]: I think that this test might randomally fail and could result in a data race.

File: txnsync/service_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+// TestMakeTransactionSyncService tests that an appropriate transaction sync service was made
+func TestMakeTransactionSyncService(t *testing.T) {
+
+	a := require.New(t)
+
+	mLogger := mockLogger{}
+	mNodeConnector := &mockNodeConnector{}
+	cfg := config.Local{}
```
Comment (tsachiherman) [inline txnsync/service_test.go:105]: Avoid using the empty config. Instead use GetDefaultLocal

**Review by tsachiherman — CHANGES_REQUESTED** [review]
Looks good. Few small requests.

File: txnsync/service_test.go (hunk 1 above)
Comment (AlgoStephenAkiki) [reply txnsync/service_test.go:91]: Isn't the Events() function in mainloop.go:124 called before any of the channel switch statements?  The Stop() function also waits for the goroutine to run right?

**Review by tsachiherman — APPROVED** [review]

**Pull request merged**